project2-beat-that/script.js

```diff
@@ -1,4 +1,272 @@
+// Generate 2 dice rolls for player 1
+// > return user the 2 dice rolls
+// > Allow player 1 to choose order of 2 dice rolls
+// > return the player's choice
+
+// 2nd version
+
+var comparingCombineDiceRolls = function () {
+  var max = Math.max.apply(null, playerScores);
+  var winningPlayer = playerScores.indexOf(max);
+  return winningPlayer;
+};
+
+var randomDiceRolls = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+var diceRoll1 = 0;
+var diceRoll2 = 0;
+
+var rollState = true;
+var playerNumber = 0;
+var numPlayers = 2;
+var playerScores = Array(numPlayers);
+playerScores.fill(0);
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // If supposed to roll
+  if (rollState === true) {
+    // If user gives non-empty input, return error msg
+    if (input !== "") {
+      return 'Please roll the dice by clicking "submit"';
+    }
+
+    // Do rolling of dice
+    diceRoll1 = randomDiceRolls();
+    diceRoll2 = randomDiceRolls();
+    console.log(diceRoll1);
+    console.log(diceRoll2);
+    // if dice rolls are the same
+    if (diceRoll1 === diceRoll2) {
+      playerScores[playerNumber] = Number("" + diceRoll1 + diceRoll2);
+      var userMsg = `It's a tie, your number is ${
+        playerScores[playerNumber]
+      }. <br> Player ${playerNumber + 1} your turn next. `;
+      if (playerNumber < numPlayers - 1) {
+        playerNumber += 1;
+        return userMsg;
+      }
+      if (playerNumber === numPlayers - 1) {
+        comparingCombineDiceRolls();
+        var returnMsg = `Your number is ${playerScores[playerNumber]}.`;
+        returnMsg += `<br> The winner is Player ${
+          comparingCombineDiceRolls() + 1
+        }.`;
+        // Reset the player
+        playerNumber = 0;
+      }
+
+      return returnMsg;
+    }
+    // Set rollState to inverse
+    else {
+      rollState = false;
+    }
+
+    var userMsg = `Hi Player ${playerNumber + 1}, you have rolled ${diceRoll1}
+    for dice one and ${diceRoll2} for dice two.
+    Choose the order of the dice by entering "1" or "2".`;
+    return userMsg;
+  } else {
+    // If user gives incorrect input, return error msg
+    if (input !== "1" && input !== "2") {
+      return 'Please input either "1" or "2"';
+    }
```

Comment:
ok but hard to follow when rollstate is set to false and when it comes into these bottom conditions

can consider refactoring to remove the layers of `if` nesting